lib/Relation.php

```diff
@@ -603,21 +664,55 @@ public function first(int $limit = null): mixed
      *
      * @return TModel|array<TModel>|null
      */
-    public function last(int $limit = null): mixed
+    public function last(int $limit = null): Model|array|null
     {
-        $this->limit($limit ?? 1);
+        $models = $this->firstOrLast($limit, false);
+
+        return isset($limit) ? $models : $models[0] ?? null;
+    }
 
-        if (array_key_exists('order', $this->options)) {
-            if (str_contains($this->options['order'], implode(' ASC, ', $this->table()->pk) . ' ASC')) {
+    /**
+     * Reverses the ordering clause if specified; otherwise, sorts in descending order by primary key
+     * as returning in ascending order is the default
+     *
+     * @return array<TModel>
```

Comment:
Does this pass static analysis..? Your docblock return type is out of sync with the actual return type.

Reply:
Yes it did; it shouldn't have, so maybe there's a bug in the static analyzer? Fixed the return type in the latest commit